### PR TITLE
Add `WorldBuilderCommandProcessor` and pull out methods from `Application#main` 

### PR DESCRIPTION
**Pull request by tsukiy0**
* `WorldBuilderCommandProcessor` takes in `WorldBuilderCommands` (dimensions, alive cell locations and build) in order to build a world
* Add `Application#buildWorldFromUserInput` for managing the logic of glueing together the `Console` and `WorldBuilderCommandProcessor`
* Add `Application#runWorld` for displaying the world as it `tick`s forward to the user

### CODE REVIEW

**Comment by tsukiy0** [comment]
Those `Application` methods allow me to test the` Console` loop using mocks

File: src/main/java/io/tsukiyo/gameoflife/Application.java [hunk 1]
```diff
@@ -18,41 +17,53 @@ public static void main(String[] args) throws InterruptedException {
...
                 new SurviveRule(3),
                 new ReviveRule(3)
         ), new NeighbourLocator());
 
-        World world = null;
+        World world = buildWorldFromUserInput(console, messageGenerator, commandParser, worldBuilderCommandProcessor);
 
-        while (Objects.isNull(world)) {
-            console.write(messageGenerator.initialAliveCellLocationPromptMessage());
+        displayTickingWorldToUser(console, messageGenerator, world);
```
Comment (mrhaynesy) [inline src/main/java/io/tsukiyo/gameoflife/Application.java]: This is much nicer!
Contemplate making all these methods non static .. somehow!

Good work though 👍

**Comment by tsukiy0** [comment]
@mwpeng made an important point that how I interact with the user to initialize a world through the `Console`(*shell*) is leaking into my *core*

let me explain,

With the console, users are initializing the `World` in a multi-step process (dimensions -> alive cell 
locations -> build) with commands via the `WorldBuilderCommandProcessor`

This multi-step process is just one of many ways a user can potentially initialize a `World`. In other words, the construction of a `World` using `WorldBuilder` (`WorldBuilder#setAliveCellsAt` and `WorldBuilder#build`) is tightly coupled to this multi-step process, when in reality, the core does not care, it just expects 4 things to build a `World`, the `Dimensions`, `Coordinate`s of alive cells, `Rule`s and `NeighbourLocator`.

e.g. What if I were to accept the initial state of the world as a file? is this multistep process still relevant? probably not.

I tackle this issue in #22

File: src/main/java/io/tsukiyo/gameoflife/Application.java (hunk 1 above)
Comment (tsukiy0) [reply src/main/java/io/tsukiyo/gameoflife/Application.java]: ye i can actually test my loops now, which was a huge question mark for me before

**Pull request merged**